File: linked_list/linked.py
# Node class
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None

# Linked list creation
head = Node(10)
head.next = Node(20)
head.next.next = Node(30)

# Traverse
current = head
while current:
    print(current.data)
    current = current.next

File: linked_list/linked.pas
program LinkedListExample;
type
  PNode = ^Node;
  Node = record
    data: integer;
    next: PNode;
  end;

var
  head, second: PNode;

begin
  New(head);
  head^.data := 10;
  New(second);
  second^.data := 20;
  head^.next := second;
  second^.next := nil;

  { Traverse }
  var current: PNode;
  current := head;
  while current <> nil do
  begin
    writeln(current^.data);
    current := current^.next;
  end;
end.
